host-controller/src/main/java/org/jboss/as/domain/controller/operations/coordination/OperationRouting.java

```diff
@@ -97,6 +123,23 @@ private static OperationRouting determineRouting(final ModelNode operation, fina
         return result;
     }
 
+    private static PathAddress findSubsystemRootAddress(PathAddress address) {
+        PathAddress result = null;
+        int size = address.size();
+        if (size > 1) {
+            int subsystemKey = Integer.MAX_VALUE;
+            String firstKey = address.getElement(0).getKey();
+            if (HOST.equals(firstKey) || PROFILE.equals(firstKey)) {
```

Comment:
what about a subsystem operation beneath a DEPLOYMENT?

Reply:
@jmesnil This logic deals with the steps provided by the user, so if your concern is a step added by another step, then this logic doesn't apply to those.

Also, this logic executes on an HC, and /deployment=foo/subsystem=bar registration would  be on the server. So the user-provided address would be /host=x/server=y/deployment=foo/subsystem=bar. And that would result in a 'true' at https://github.com/wildfly/wildfly-core/blob/master/host-controller/src/main/java/org/jboss/as/domain/controller/operations/coordination/OperationRouting.java#L87 so long as the HC knows about host 'x' (if it's the DC) and server 'y' (if it's host 'x').

If L87 was false then we have an unknown remote process and we want to fail. With my change here it will drop into the 'else if (compositeStep)' block. The logic there won't match  /host=x/server=y/deployment=foo/subsystem=bar which is what we want.

A flaw I realized while writing this is it will match /host=invalid_typo/subsystem=jmx. The request will still fail later anyway so I don't think that's a big deal. But if it's easy enough to correct I'll correct it.